security/openpgp.ksy

```diff
@@ -0,0 +1,508 @@
+        type: u1
+        # enum: TODO switch?
+      - id: signature_type
+        type: u1
+        # enum: TODO 5.2.1
+      - id: public_key_algorithm
+        type: u1
+        enum: public_key_algorithms
+      - id: hash_algorithm
+        type: u1
+        enum: hash_algorithms
+      - id: hashed_subpacket_length
+        type: u2
+      - id: hashed_subpackets
+        type: subpackets
+        size: hashed_subpacket_length
+      - id: unhashed_subpacket_length
+        type: u2
+      - id: unhashed_subpackets
+        type: subpackets
+        size: unhashed_subpacket_length
+      - id: left_signed_hash
+        type: u2
+      - id: rsa_n
+        type: u2
+      - id: signature
+        size-eos: true
+    
+  secret_key_packet:
+    seq:
+      - id: public_key
+        type: public_key_packet
+      - id: string_to_key
+        type: u1
+      - id: symmetric_encryption_algorithm
+        type: u1
+        enum: symmetric_key_algorithm
+        if: string_to_key >= 254
+      - id: more
+        size-eos: true
+        
+  subpackets:
+    seq:
+      - id: subpacketss
+        type: subpacket
+        repeat: eos
+  
+  subpacket:
+    seq:
+      - id: length
+        type: subpacket_length
+      - id: subpacket_type
+        type: u1
+        enum: subpacket_types
+      - id: content
+        size: length.length - 1
+        type: 
+          switch-on: subpacket_type
+          cases:
+            subpacket_types::signature_creation_time: signature_creation_time
+            subpacket_types::issuer: issuer
+            subpacket_types::key_expiration_time: key_expiration_time
+            subpacket_types::preferred_hash_algorithms: preferred_hash_algorithms
+            subpacket_types::preferred_compression_algorithms: preferred_compression_algorithms
+            subpacket_types::signature_expiration_time: signature_expiration_time
+            subpacket_types::exportable_certification: exportable_certification
+            subpacket_types::revocable: revocable
+            subpacket_types::trust_signature: trust_signature
+            subpacket_types::regular_expression: regular_expression
+            subpacket_types::revocation_key: revocation_key
+            subpacket_types::notation_data: notation_data
+            subpacket_types::key_server_preferences: key_server_preferences
+            subpacket_types::preferred_key_server: preferred_key_server
+            subpacket_types::primary_user_id: primary_user_id
+            subpacket_types::policy_uri: policy_uri
+            subpacket_types::key_flags: key_flags
+            subpacket_types::signers_user_id: signers_user_id
+            subpacket_types::reason_for_revocation: reason_for_revocation
+            subpacket_types::features: features
+            subpacket_types::signature_target: signature_target
+            subpacket_types::embedded_signature: embedded_signature
+      
+  subpacket_length:
+    -webide-representation: '{length}'
```

Comment:
Is it a variable length integer? Please check ```common``` directory.

Reply:
Nope, it's different.